core/arch/arm/kernel/thread_a64.S

```diff
@@ -229,13 +229,16 @@ FUNC thread_rpc , :
 	str	x2, [x19, #THREAD_CTX_REGS_SP]
 	ldr	x20, [sp]	/* Get pointer to rv[] */
 	mov	sp, x0		/* Switch to tmp stack */
+	load_wregs x20, 0, 21, 23	/* Load rv[] into w20-w22 */
```

Comment:
Why is this needed (loading the registers before the call to thread_state_suspend())?

Reply:
Because `thread_state_suspend()` calls `virt_unset_guest()` which removes all guest mappings, so we can't read from `rv[]` after that.

Reply:
Aha, thanks. Yes a comment would be good.

I guess you need the same change in `core/arch/arm/kernel/thread_a32.S`.

Reply:
Sure, will add comment. 

As for 32-bit version... Virtualization is not supported on 32 bits right now. I can make similar changes to `thread_a32.S`, but this will be blind shot, because I can't test it with virtualization enabled.